spec/lib/checklist_syntax_migrator_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ChecklistSyntaxMigrator do
+  before do
+    SiteSetting.allow_uncategorized_topics = true
+  end
+  let(:topic) { Fabricate(:topic) }
+  let(:post_args) do
+    { user: topic.user, topic: topic }
+  end
+
+  def post_with_body(body)
+    args = post_args.merge(raw: body)
+    Fabricate.build(:post, args)
+  end
+
+  it "replaces instances of the old checkbox instance, with the new syntax" do
+    body = "[-] 1\n[_] 2\n[*] 3\n[\\*] 4\n[\*] 5\n[*\] 6"
```

Comment:
3, 5, and 6 (`"[*]"`, `"[\*]"`, and `"[*\]"`) test exactly the same thing as single escape characters (`\`) in that string are removed during parsing of this .rb file and don't make it to the `ChecklistSyntaxMigrator`.